src/server/semester/semester.service.ts

```diff
@@ -59,4 +62,150 @@ export class SemesterService {
+          const springInstance: CourseInstance = courseIdToSpringInstance[
+            fallInstance.course.id
+          ];
+          return {
+            ...new CourseInstance(),
+            offered: springInstance.offered === OFFERED.RETIRED
+              ? OFFERED.RETIRED : OFFERED.BLANK,
+            preEnrollment: null,
+            studyCardEnrollment: null,
+            actualEnrollment: null,
+            course: fallInstance.course,
+            facultyCourseInstances: [],
+            meetings: [],
+          };
+        });
+
+      // Create new fall semester
+      const fallSemester = new Semester();
+      // NOTE: academic year in Semester entity is actually the calendar year
+      fallSemester.academicYear = (newAcademicYear - 1).toString();
+      fallSemester.term = TERM.FALL;
+      fallSemester.courseInstances = newFallInstances;
+
+      fallSemester.nonClassEvents = existingFallSemester.nonClassEvents
+        .map((nce) => ({
+          ...new NonClassEvent(),
+          nonClassParent: nce.nonClassParent,
+          private: nce.private,
+          meetings: [],
+        }));
+
+      fallSemester.absences = existingFallSemester.absences
+        .map((absence) => ({
+          ...new Absence(),
+          faculty: absence.faculty,
+          type: ABSENCE_TYPE.PRESENT,
+        }));
+
+      // Note that this also saves the nested entities because of
+      // the cascade set on semester
+      await this.semesterRepository.save(fallSemester);
+
+      const newSpringInstances = springInstances
+        .map((springInstance) => ({
+          ...new CourseInstance(),
+          offered: springInstance.offered === OFFERED.RETIRED
+            ? OFFERED.RETIRED : OFFERED.BLANK,
+          preEnrollment: null,
+          studyCardEnrollment: null,
+          actualEnrollment: null,
+          course: springInstance.course,
+          facultyCourseInstances: [],
+          meetings: [],
+        }));
+
+      // Create new spring semester
+      const springSemester = new Semester();
+      springSemester.academicYear = newAcademicYear.toString();
+      springSemester.term = TERM.SPRING;
+      springSemester.courseInstances = newSpringInstances;
+
+      springSemester.nonClassEvents = existingSpringSemester.nonClassEvents
+        .map((nce) => ({
+          ...new NonClassEvent(),
+          nonClassParent: nce.nonClassParent,
+          private: nce.private,
+          meetings: [],
+        }));
+
+      springSemester.absences = existingSpringSemester.absences
+        .map((absence) => ({
+          ...new Absence(),
+          faculty: absence.faculty,
+          type: ABSENCE_TYPE.PRESENT,
+        }));
+
+      await this.semesterRepository.save(springSemester);
+    }
+    return null;
+  }
+
+  public async onApplicationBootstrap(): Promise<void> {
+    const today = new Date();
+    if (today.getMonth() === 5) {
+      await this.addAcademicYear(today.getFullYear() + 1);
+    }
+  }
```

Comment:
Oh I see, ok thanks for explaining this. I wrapped the call to `addAcademicYear` in a `try...catch` so that we could log any errors and added other logs so that we could see where in the process we are in adding the new academic year. [bc9c831](https://github.com/seas-computing/course-planner/pull/496/commits/bc9c831e0468e0a628af5408b6cbe272ab9f7dcd)

Reply:
Ah, those are being generated in the [`ConfigService`](https://github.com/seas-computing/course-planner/blob/main/src/server/config/config.service.ts#L310), but you're right that those are pretty excessive. We could wrap that line with something like:

```ts
// config.service.ts Line 310
if (this.isProduction) {
  console.warn(`"${logLevel}" is not a valid LOG_LEVEL. Defaulting to "error"`);
}
```

Reply:
Ok thanks! I added the conditional in [f8d3f98](https://github.com/seas-computing/course-planner/pull/496/commits/f8d3f9849457952003f920006a4ca8d215501170) so that we do not see these messages while running the tests.